### ISSUE TITLE
Set up PR vs production configs for the deploy action

### ISSUE THREAD

**Issue by jbusecke** [leap-stc/data-management#32]
We are currently building the full recipe for every commit (if a label is set). 

I think we should have some things set in two different config jsons based on the event type:

- Push to main:
    - Prune set to false
    - Set a permanent directory in `leap-persistent-ro`for each recipe and subfolders for each action run
- PR:
    - Prune set to true
    - Write resulting data to temporary storage (e.g. `leap-scratch`)

**Comment by cisaacstern** [comment]
This could be achieved with conditionals in the workflow file (and no changes needed to the action). Or it could be a PR to the action. Either seems doable.

**Comment by jbusecke** [comment]
I would think the first option is more flexible right now? I guess we really need some more users of the action to see how we might properly generalize this?

**Comment by cisaacstern** [comment]
> I would think the first option is more flexible right now?

Agreed. A relatively manageable path forward would be to use a github workflow step that conditionally exports either env vars (or workflow step outputs) based on the github event type, which can then be inserted into the pangeo-forge-runner JSON config.

**Comment by cisaacstern** [comment]
In #33, I currently have the following parts of this working:

_Push to main:_

  - [x] Prune set to false
  - [ ] Set a permanent directory in leap-persistent-rofor each recipe and subfolders for each action run
  
_PR:_
  - [x] Prune set to true
  - [ ] Write resulting data to temporary storage (e.g. leap-scratch)
